### ISSUE TITLE
When to call scheduler.step? 

### ISSUE THREAD

**Issue by benihime91** [ildoonet/pytorch-gradual-warmup-lr#22]
scheduler.step.is call after each batch or epoch?

**Comment by saniazahan** [comment]
indeed, I am also curious as we usually call optim.step() after each batch
